### PR TITLE
View specs

### PR DESCRIPTION
**Pull request by Naledi-Dikgale**
### In this PR we did the following

- [ ] Used Capybara to write integration tests for each view in my project.
```
- User index page
- user show page
- User post index page
- Post show page
```

- [ ] Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.

- screenshot before fixing n+1

beforen+1

- screenshot after fixing n+1

Aftern+1

### CODE REVIEW

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
Hi @Naledi-Dikgale 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- [ ] You have submitted a project, that does not have a significant contribution from  partner(s)

- [ ] This milestone is a [pair-programming activity](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#work-and-submission-mode).

- [ ] Please ensure that everyone has contributed equally to the project you're reviewing.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/views/users/users_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+RSpec.describe 'User', type: :feature, js: false do
+  before :each do
+    @user = User.create(name: 'Naledi Dikgale', photo: 'https://e0.pxfuel.com/wallpapers/820/800/desktop-wallpaper-cute-black-girls-black-cartoon-characters-thumbnail.jpg',
+                        bio: 'Undisputed Queen of DIY')
+    @first_post = Post.create(author: @user, title: 'Naledi', text: 'How to make a table')
+    @comment = Comment.create(post: @first_post, author: @user, text: 'Nice post!')
+
+    visit "/users/#{@user.id}/posts"
+  end
```
Comment (OyePriscilla) [inline spec/views/users/users_index_spec.rb]: - [ ] The tests written in this file suppose to be for `users show` page according to the file name I think, the `url` added is pointing at the` post homepage` not `users show' page`. 👍 
- [ ] Kindly implement all the tests for user_show page as indicated in the requirement. 👍

File: spec/views/users/users_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+require 'capybara/rspec'
+
+base_url = 'http://localhost:3000'
+
```
Comment (OyePriscilla) [inline spec/views/users/users_show_spec.rb]: - [ ] If we go by the content of this file, I think the file name should be `users_index_spec.rb` not `users_show_spec.rb`. 👍 
- [ ] Please kindly add the missing test as well. ℹ️ 

Untitled

File: spec/views/posts/posts_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    @first_post = Post.create(author: @user, title: 'Naledi', text: 'How to make a table')
+    @comment = Comment.create(post: @first_post, author: @user, text: 'Nice post!')
+
+    visit "/users/#{@user.id}/posts"
+  end
+
+  describe 'Post page' do
+    it "shows the user's profile" do
+      expect(page).to have_css("img[src*='https://e0.pxfuel.com/wallpapers/820/800/desktop-wallpaper-cute-black-girls-black-cartoon-characters-thumbnail.jpg']")
+    end
```
Comment (OyePriscilla) [inline spec/views/posts/posts_index_spec.rb]: - [ ] Please kindly add the following missing tests as well in accordanc to the project requirement. ℹ️

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Naledi-Dikgale and @lugard1 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Used Capybara to write integration tests ✔️ 
Had featured tests for `Post index` page ✔️ 
Had featured tests for` Post show` page ✔️ 
Had featured tests for `User index` page ✔️ 
Had featured tests for `User show` page ❎ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/users/users_show_spec.rb [hunk 4]
```diff
@@ -19,8 +19,31 @@
...
+    end
+
     it 'should display the number of posts' do
       users.each { |user| expect(page).to have_content("Number of posts: #{user.posts.count}") }
     end
+
+    it 'should display user bio' do
+      users.each { |user| expect(page).to have_content(user.bio) }
+    end
+
```
Comment (Ismailco) [inline spec/views/users/users_show_spec.rb:30]: - [ ] Kindly make sure to implement tests that assert you can see a button that lets you view all of a user's posts.

File: spec/views/users/users_index_spec.rb [hunk 5]
```diff
@@ -40,5 +40,10 @@
       expect(page).to have_content(user_post.comments.count)
       expect(page).to have_content(user_post.likes.count)
     end
+
```
Comment (Ismailco) [inline spec/views/users/users_index_spec.rb]: - [ ] This is a test for the `Posts` index page as the previous reviewer mentioned kindly consider moving these tests to the `posts_index_spec.rb` file instead or remove them altogether

File: spec/views/posts/posts_index_spec.rb [hunk 6]
```diff
@@ -35,10 +35,28 @@
       expect(page).to have_content(@comment.text)
     end
```
Comment (Ismailco) [inline spec/views/posts/posts_index_spec.rb]: - [ ] These tests are the same as the ones you created in the `users_index_spec.rb` kindly make sure to not repeat the same tests in different files, instead you should keep these tests and modify the `users_index_spec.rb` to test the `Users` index page

File: spec/views/users/users_index_spec.rb (hunk 5 above)
Comment (Ismailco) [inline spec/views/users/users_index_spec.rb]: - [ ] As the previous reviewer suggested kindly consider adding tests for the `User` index page by modifying this file and removing unrelated tests. here are the required tests for the user index page 👇🏼. You can see in the before method on line 10 that you're visiting the posts index page instead ` visit "/users/#{@user.id}/posts"`, it should be   `visit "/users"`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Naledi-Dikgale, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Naledi-Dikgale and @lugard1,

✔️ Well-written and passing tests

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**